applications/luci-app-shadowsocks-libev/htdocs/luci-static/resources/view/shadowsocks-libev/instances.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+'require form';
+'require uci';
+'require fs';
+'require network';
+'require rpc';
+'require shadowsocks-libev as ss';
+
+var conf = 'shadowsocks-libev';
+var cfgtypes = ['ss_local', 'ss_redir', 'ss_server', 'ss_tunnel'];
+
+var callServiceList = rpc.declare({
+	object: 'service',
+	method: 'list',
+	params: [ 'name' ],
+	expect: { '': {} }
+});
+
+return L.view.extend({
+	render: function(stats) {
+		var m, s, o;
+
+		m = new form.Map(conf,
+			_('Local Instances'),
+			_('Instances of shadowsocks-libev components, e.g. ss-local, \
+			   ss-redir, ss-tunnel, ss-server, etc.  To enable an instance it \
+			   is required to enable both the instance itself and the remote \
+			   server it refers to.'));
+
+		s = m.section(form.GridSection);
+		s.addremove = true;
+		s.cfgsections = function() {
+			return this.map.data.sections(this.map.config)
+				.filter(function(s) { return cfgtypes.indexOf(s['.type']) !== -1; })
+				.map(function(s) { return s['.name']; });
+		};
+		s.sectiontitle = function(section_id) {
+			var s = uci.get(conf, section_id);
+			return (s ? s['.type'] + '.' : '') + section_id;
+		};
+		s.renderSectionAdd = function() {
+			var el = form.GridSection.prototype.renderSectionAdd.apply(this, arguments),
+				optionEl = [E('option', { value: '_dummy' }, [_('-- instance type --')])];
+			cfgtypes.forEach(function(t) {
+				optionEl.push(E('option', { value: t }, [t.replace('_', '-')]));
+			});
+			var selectEl = E('select', {
+				class: 'cbi-input-select',
+				change: function(ev) {
+					ev.target.parentElement.nextElementSibling.nextElementSibling
+						.toggleAttribute('disabled', ev.target.value === '_dummy');
+				}
+			}, optionEl);
+			el.lastElementChild.setAttribute('disabled', '');
+			el.prepend(E('div', {}, selectEl));
+			return el;
+		};
+		s.handleAdd = function(ev) {
```

Comment:
#3256 just got merged.  Let's keep the func prototype the same.  It's easier to read

Reply:
Fixed.